ext/standard/pack.c

```diff
@@ -104,8 +104,47 @@ static void php_pack(zval *val, size_t size, int *map, char *output)
 }
 /* }}} */
 
+
+/* {{{ php_pack_is_little_endian
+ */
+static int php_pack_is_little_endian()
+{
+	int machine_endian_check = 1;
+	
+	return ((char *)&machine_endian_check)[0];
+}
+/* }}} */
+
+
+/* {{{ php_pack_memcpy
+ */
+static void php_pack_memcpy(int is_little_endian, void * dst, void * src, size_t size)
+{
+	int is_sys_little_endian = php_pack_is_little_endian();
+	
+	if (is_little_endian == is_sys_little_endian) {
+		/* 
+		  System and input uses same endian, simply perform memcpy 
+		*/
+		memcpy(dst, src, size);
+	} else {
+		/*
+		  Copy to dst from last to first 
+		*/
+		src = src + size - 1;
+		do {
+			*(char *) dst = *(char *)src;
+			++dst;
```

Comment:
arithmethic for void pointer is illegal, gcc allows it  just as a extension. I think  you'd better cast it before ++